abnf/odata-abnf-construction-rules.txt

```diff
@@ -928,54 +928,61 @@ primitiveValue = booleanValue
-
-string           = SQUOTE *( SQUOTE-in-string / pchar-no-SQUOTE ) SQUOTE
+binaryLiteral = "binary" SQUOTE binaryValue SQUOTE
+binaryValue   = *(4base64char) [ base64b16  / base64b8 ]
+base64b16     = 2base64char ( %s"A" / %s"E" / %s"I" / %s"M" / %s"Q" / %s"U" / %s"Y" / %s"c" / %s"g" / %s"k" / %s"o" / %s"s" / %s"w" / %s"0" / %s"4" / %s"8" )   [ "=" ]
+base64b8      = base64char ( %s"A" / %s"Q" / %s"g" / %s"w" ) [ "==" ]
+base64char    = ALPHA / DIGIT / "-" / "_"
+
+boolean = "true" / "false"
+
+decimalLiteral = [ SIGN ]  1*DIGIT [ "." 1*DIGIT ] [ "e" [ SIGN ]  1*DIGIT ] / nanInfinity
+decimalValue   = ["+"/"-"] 1*DIGIT [ "." 1*DIGIT ] [ "e" ["+"/"-"] 1*DIGIT ] / nanInfinity
+doubleLiteral  = decimalLiteral ; IEEE 754 binary64 floating-point number (15-17 decimal digits)
+doubleValue    = decimalValue   ; IEEE 754 binary64 floating-point number (15-17 decimal digits)
+singleLiteral  = decimalLiteral ; IEEE 754 binary32 floating-point number (6-9 decimal digits)
+singleValue    = decimalValue   ; IEEE 754 binary32 floating-point number (6-9 decimal digits)
+nanInfinity    = %s"NaN" / %s"-INF" / %s"INF"
+
+guid = 8HEXDIG "-" 4HEXDIG "-" 4HEXDIG "-" 4HEXDIG "-" 12HEXDIG
+
+byte         = 1*3DIGIT            ; numbers in the range from 0 to 255
+sbyteLiteral = [ SIGN ]  1*3DIGIT  ; numbers in the range from -128 to 127
+sbyteValue   = ["+"/"-"] 1*3DIGIT  ; numbers in the range from -128 to 127
+int16Literal = [ SIGN ]  1*5DIGIT  ; numbers in the range from -32768 to 32767
+int16Value   = ["+"/"-"] 1*5DIGIT  ; numbers in the range from -32768 to 32767
+int32Literal = [ SIGN ]  1*10DIGIT ; numbers in the range from -2147483648 to 2147483647
+int32Value   = ["+"/"-"] 1*10DIGIT ; numbers in the range from -2147483648 to 2147483647
+int64Literal = [ SIGN ]  1*19DIGIT ; numbers in the range from -9223372036854775808 to 9223372036854775807
+int64Value   = ["+"/"-"] 1*19DIGIT ; numbers in the range from -9223372036854775808 to 9223372036854775807
+
+stringLiteral    = SQUOTE *( SQUOTE-in-string / pchar-no-SQUOTE ) SQUOTE
 SQUOTE-in-string = SQUOTE SQUOTE ; two consecutive single quotes represent one within a string literal
 
-dateValue = year "-" month "-" day
+date = year "-" month "-" day
 
-dateTimeOffsetValue      = year "-" month "-" day "T" timeOfDayValue      ( "Z" / SIGN hour ":"   minute )
-dateTimeOffsetValueInUrl = year "-" month "-" day "T" timeOfDayValueInUrl ( "Z" / SIGN hour COLON minute )
+dateTimeOffsetLiteral = date "T" timeOfDayLiteral ( "Z" / SIGN      hour COLON minute )
+dateTimeOffsetValue   = date "T" timeOfDayValue   ( "Z" / ("+"/"-") hour ":"   minute )
 
-duration      = [ "duration" ] SQUOTE durationValue SQUOTE
-durationValue = [ SIGN ] "P" [ 1*DIGIT "D" ] [ "T" [ 1*DIGIT "H" ] [ 1*DIGIT "M" ] [ 1*DIGIT [ "." 1*DIGIT ] "S" ] ]
+durationLiteral = [ "duration" ] SQUOTE durationValue SQUOTE
+durationValue = [ "-" ] "P" [ 1*DIGIT "D" ] [ "T" [ 1*DIGIT "H" ] [ 1*DIGIT "M" ] [ 1*DIGIT [ "." 1*DIGIT ] "S" ] ]

```

Comment:
Incompatible change?

Reply:
Not really, see CSDL [14.3.6 Duration](https://docs.oasis-open.org/odata/odata-csdl-json/v4.01/odata-csdl-json-v4.01.html#sec_Duration)
> The value MUST conform to type xs:dayTimeDuration, see [[XML‑Schema‑2]](https://docs.oasis-open.org/odata/odata-csdl-json/v4.01/odata-csdl-json-v4.01.html#BMXMLSchema2), [section 3.4.27](http://www.w3.org/TR/xmlschema11-2/#dayTimeDuration).
XML Schema does not allow a plus sign for positive durations, it only allows a minus sign for negative durations.